### PR TITLE
Move to C++17

### PR DESCRIPTION
**Pull request by gvoskuilen**

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -32,12 +32,12 @@ AC_CACHE_SAVE
...
 
 dnl Work out the C++ standard which we are going to use
-AX_CXX_COMPILE_STDCXX_1Y
-AX_CXX_COMPILE_STDCXX_0X
+AX_CXX_COMPILE_STDCXX_17(noext)
 
-AS_IF([test "x$found_cxx1y" = "xyes"],
-        [CXXFLAGS="$CXXFLAGS $SST_CXX1Y_FLAGS"],
-        [CXXFLAGS="$CXXFLAGS $SST_CXX0X_FLAGS"])
+CXXFLAGS="$CXXFLAGS -std=c++17"
```
Comment (jpkenny) [inline configure.ac:37]: I think you should just leave this line out. I believe the macro will just redefine CXX (CXX="$CXX $switch") and will detect different switches for different compilers. Probably not a problem for the compilers we support, but it's the way it's intended to work. (Do your compile commands have duplicate -std=c++17?)

File: configure.ac [hunk 2]
```diff
@@ -32,12 +32,12 @@ AC_CACHE_SAVE
 LT_INIT([shared disable-static dlopen])
 
 dnl Work out the C++ standard which we are going to use
-AX_CXX_COMPILE_STDCXX_1Y
-AX_CXX_COMPILE_STDCXX_0X
+AX_CXX_COMPILE_STDCXX_17(noext)
```
Comment (jpkenny) [inline configure.ac:35]: If you leave off the noext argument (leave it blank) I don't think it hurts anything and could potentially make some obscure hypothetical compiler that doesn't support strict conformance work. Not a big deal.

**Review by jpkenny — COMMENTED** [review]
A couple changes to consider.

File: configure.ac (hunk 2 above)
Comment (gvoskuilen) [reply configure.ac:35]: noext is required but I will fix the macro so that it doesn't overwrite CXX

**Review by jpkenny — COMMENTED** [review]
Let's just stop AX_CXX_COMPILE_STDCXX from adding -std flag to CXX/CXXCPP and call it a day.

File: configure.ac (hunk 1 above)
Comment (gvoskuilen) [reply configure.ac:37]: That flag is required there but I am removing -std=c++17 from CXX so it's cleaner.

**Review by jpkenny — APPROVED** [review]
Looks good

**Pull request merged**